src/player.rs

```diff
@@ -1,32 +1,122 @@
 use crate::constants;
+use crate::constants::{MAP_SCALE_H, MAP_SCALE_W};
+use crate::map;
+use crate::map::Tile;
 
-const ROTATE_SPEED: f64 = 0.02;
+const ROTATE_SPEED: f64 = 0.2;
 const MOVE_SPEED: f64 = 2.5;
+const PLAYER_WIDTH: f64 = 7.0;
 
 pub struct Player {
     pub x: f64,
     pub y: f64,
-    pub angle: f64,
+    pub view_angle: f64,
+    pub move_angle: f64,
 }
 
 impl Player {
-    pub fn move_forward(&mut self) {
-        self.x += self.angle.sin() * MOVE_SPEED;
-        self.y += self.angle.cos() * MOVE_SPEED;
-    }
+    /// Moves player across the map and prevents stepping into walls.
+    /// Player collision box is a square. Its vertices are checked for collision with nearby walls.
+    pub fn move_player(&mut self, map: &map::Map) {
```

Comment:
```suggestion
    pub fn move(&mut self, map: &map::Map) {
```

the player suffix is redundant in a player struct method

Reply:
I agree, but method name "move" is illegal in Rust so I renamed it to "walk" in latest commit revision cee80e28ba6ae7588d6c8b6c5ef1d47295868f7a.
Thank you.